src/main/java/de/tum/in/www1/artemis/security/jwt/JWTFilter.java

```diff
@@ -28,23 +30,30 @@ public JWTFilter(TokenProvider tokenProvider) {
     @Override
     public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
         HttpServletRequest httpServletRequest = (HttpServletRequest) servletRequest;
-        String jwt = resolveToken(httpServletRequest);
-        if (StringUtils.hasText(jwt) && this.tokenProvider.validateTokenForAuthority(jwt)) {
-            Authentication authentication = this.tokenProvider.getAuthentication(jwt);
-            SecurityContextHolder.getContext().setAuthentication(authentication);
+        Cookie jwtCookie = WebUtils.getCookie(httpServletRequest, JWT_COOKIE_NAME);
+        if (jwtCookie != null) {
+            String jwt = jwtCookie.getValue();
+            if (StringUtils.hasText(jwt) && this.tokenProvider.validateTokenForAuthority(jwt)) {
+                Authentication authentication = this.tokenProvider.getAuthentication(jwt);
+                SecurityContextHolder.getContext().setAuthentication(authentication);
+            }
         }
         filterChain.doFilter(servletRequest, servletResponse);
     }
 
-    private String resolveToken(HttpServletRequest request) {
-        String bearerToken = request.getHeader(AUTHORIZATION_HEADER);
-        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
-            return bearerToken.substring(7);
-        }
-        String jwt = request.getParameter(AUTHORIZATION_TOKEN);
-        if (StringUtils.hasText(jwt)) {
-            return jwt;
-        }
-        return null;
+    /**
+     * Builds the cookie containing the jwt
+     * @param jwt      the token that will be used as the cookie's value
+     * @param duration the validity of the cookie
+     * @return         the response cookie that should be set containing the jwt
+     */
+    public static ResponseCookie buildJWTCookie(String jwt, Duration duration) {
+
+        return ResponseCookie.from(JWT_COOKIE_NAME, jwt).httpOnly(true) // Must be httpOnly
+                .sameSite("Lax") // Must be Lax to allow navigation links to Artemis to work
+                .secure(false) // Should only be secure when using https, otherwise in environments using http (cypress) browsers don't set the cookie
```

Comment:
I agree, let's use "Strict"

Reply:
@julian-christl we're definitely not leaving secure=false, I was just testing if secure=true was the cause for the cypress tests to fail. It's not, it must be a separate issue but I can't figure out why they're failing. Either way, I'll set the cookie to secure.

About the cypress tests: I think the issue is that the cookie isn't being set, since all requests return a 401. The cookie is set in local cypress tests and is set when using the test environments, so if you have an idea why it's not set in the cypress tests I'd appreciate any help!